pkg/localcontroller/manager/dataset.go

```diff
@@ -105,6 +110,14 @@ func (dm *DatasetManager) Insert(message *gmclient.Message) error {
 		return err
 	}
 
+	credential := dataset.ObjectMeta.Annotations[Credential]
+	if credential != "" {
+		dataset.Storage = storage.Storage{}
```

Comment:
this will reset the field `Storage` of `dataset` multiple times

Reply:
fix it